doc/user/content/transform-data/patterns/window-functions.md

```diff
@@ -8,8 +8,10 @@ menu:
     parent: 'sql-patterns'
 ---
 
-Some query patterns that are commonly used in batch are tricky to support in streaming due to the unbounded nature of the input data. Window functions, such as `LAG`, `LEAD`, `ROW_NUMBER`, `FIRST_VALUE`, are one of those patterns.
-As is, Materialize doesn't handle window functions efficiently for large partition sizes: when an input record is added, removed or changed, the system recomputes the results for the entire partition that the changed record belongs to. For this reason, we recommend using the alternative approaches described in this page whenever possible, until window function support is refactored to provide better performance.
+Some query patterns that are commonly used in batch systems are tricky to support with continuously arriving input data. Window functions, such as `LAG`, `LEAD`, `ROW_NUMBER`, `FIRST_VALUE`, are one of those patterns.
```

Comment:
Feel free to rewrite, but we should open with a sentence that clearly states we do support window functions. 
```suggestion
Materialize implements a wide range of window functions to support a variety of complex query patterns. 
Some query patterns that are commonly used in batch systems are tricky to support with continuously arriving input data. Window functions, such as `LAG`, `LEAD`, `ROW_NUMBER`, `FIRST_VALUE`, are one of those patterns.
```

Reply:
Thanks! I made the change.